rust/rpc/src/lib.rs

```diff
@@ -271,6 +273,10 @@ impl<W: Write + Send> RpcLoop<W> {
                     let json = match self.reader.next(&mut stream) {
                         Ok(json) => json,
                         Err(err) => {
+                            if self.peer.0.is_blocked.load(Ordering::Relaxed) {
```

Comment:
Although it's unlikely to be a problem in practice, I think we should use `Ordering:SeqCst` here; I think as written there's still a theoretical risk of deadlock.